core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -585,15 +587,25 @@ public void convertToLeader(int leaderEpochVal) {
             if (this.leaderEpoch != leaderEpochVal) {
                 leaderEpoch = leaderEpochVal;
             }
-            // Reset readOffset, so that it is set in next run of RLMTask
+            // Reset copied and log-start offset, so that it is set in next run of RLMTask
             copiedOffsetOption = OptionalLong.empty();
+            logStartOffsetOption = OptionalLong.empty();
         }
 
         public void convertToFollower() {
             leaderEpoch = -1;
         }
 
-        private void maybeUpdateReadOffset(UnifiedLog log) throws RemoteStorageException {
+        private void maybeUpdateLogStartOffset(UnifiedLog log) throws RemoteStorageException {
+            if (!logStartOffsetOption.isPresent()) {
+                logStartOffsetOption = OptionalLong.of(findLogStartOffset(topicIdPartition, log));
+                logStartOffsetOption.ifPresent(offset -> updateRemoteLogStartOffset.accept(topicIdPartition.topicPartition(), offset));
+                logger.info("Found the log start offset: {} for partition: {} after becoming leader, leaderEpoch: {}",
+                        logStartOffsetOption, topicIdPartition, leaderEpoch);
```

Comment:
The log is not completely correct. Since the RLMTask will be run periodically, the log start offset update might happen not because of leadership change (ex: record deletion).

Reply:
Not clear on this comment. `maybeUpdateLogStartOffset` will be invoked only when `logStartOffsetOption` is empty. The `logStartOffsetOption` will be empty when it becomes the leader.

Reply:
Addressed your comment by removing the offset holder (logStartOffsetOption) and used a boolean (isLogStartOffsetUpdatedOnBecomingLeader) for code clarity.

Reply:
> The logStartOffsetOption will be empty when it becomes the leader.

You're right! Thanks for the update.